### PR TITLE
[CR] Wallet item

### PR DESCRIPTION
**Pull request by Scrivener07**
#### Summary
SUMMARY: Content "Adds wallet item for holding cash & id cards."

#### Purpose of change
A wallet for organizing the many id & cash cards found in CDDA. This was something that just feels right as an item almost everyone won't leave their homes without.

#### Describe the solution
"Organizing" in the practical sense means being able to pickup and drop your various cards with a single action.
It can also act as a "favorites group" where you only store the cards you want to carry around with you.
Like a keychain, it keeps things together.

#### Describe alternatives you've considered
I've tried using smaller bags and a purse but it doesn't have nearly the same flavor appeal.

#### Testing
I have tested this item and its craftability using _Play Now!_ characters and the debug menus.

#### Additional context
Someone on the discord thought my idea was cool if that counts for anything :P

### CODE REVIEW

**Comment by Scrivener07** [comment]
There is a lot I dont know about mod/dev in this game. So any critical feedback is welcome. I dont have full confidence in the pocket & item volume/weights. Im not sure about the "looks like" being the plastic bag. Im not sure that I choose all the right categories, types, and files to add things too. Im more than happy to go back and make changes.

**Comment by UmbralReaper** [comment]
I'd suggest adding leather & plastic variants of the wallet to the game. Not everybody wants an unfashionable duct tape wallet :).

**Comment by DialoMalison** [comment]
I agree with UmbralReaper about the leather & plastic variants. Also, consider adding these to various spawn lists (houses, clothing stores, corpses, etc.) to represent the millions of wallets that were floating around pre-cataclysm New England.

**Comment by Scrivener07** [comment]
Thanks for the feedback. Im not sure about the spawn lists yet but the rest should be easy. These are some of the variants I thought of.

- Common Wallet
  - Name: wallet
  - Pocket: Cards Only

- Duct Tape Wallet
  - Name: duct tape wallet
  - Pocket: Cards Only
  - Craftable: Fabrication 1

- Leather Wallet
  - Name: leather wallet
  - Pocket: Cards Only
  - Craftable: Tailor 2

- Large Wallet
  - Name: large wallet
  - Pocket: Cards Only
  - Pocket: Coin Purse
  - Pocket: Small Items

- Stylish Wallet
  - Name: stylish wallet
  - Pocket: Cards Only
  - Pocket: Coin Purse
  - Stylish

- Travel Wallet
  - Name: travel wallet
  - Pocket: Cards Only
  - Pocket: Coin Purse
  - Pocket: Small Items
  - Wearable: Neck

**Comment by DialoMalison** [comment]
Looks to me like you have got all the bases covered.  I like the travel wallet, it's a nice touch. Looking forward to seeing them in the game.

**Comment by 8street** [comment]
Maybe [item spawn](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/ITEM_SPAWN.md) manual will help for adding spawn lists. Also I just mention this topic #42097.

**Comment by Entity-y** [comment]
I was thinking in addition to the wallets you've made to also include a late game surival expandable wallet of some kind made with leather and neoprene that stretches to hold far more cards. A wallet that fat would sure be encumbering.

**Comment by Entity-y** [comment]
To add to the flavor, wallets should sometimes include pictures, sd cards, bobby pins. As light humor maybe have a rare instance of including condoms or even a ramen flavor packet.

**Comment by LaVeyanFiend** [comment]
You could also put the various faction currencies in ``items/generic/currency.json`` in as another option to put in the wallet.

**Comment by Fris0uman** [comment]
By the way you should relove the [CR] from the title once you think this is ready or it'll never get merged.

**Comment by LyleSY** [comment]
Looking at the  JSON validation check, your generic.json data needs to be copied, pasted here https://dev.narc.ro/cataclysm/format.html , the "Lint" button clicked, and then that result copied and pasted back into generic.json , I think because you have a bunch of spaces in there. This is a pain I know well.

**Comment by Scrivener07** [comment]
Thanks for all the feedback from everyone. I have not worked on this since my last comment, but plan to pick it up again soon. I left off learning how to implement the new pocket data and re-learning basic unit conversions. I need to make sure each of these wallets can fit into all the appropriate parent pockets like **common wallet** into rear jeans pocket, but not **large wallets**. 

I thought to add the **travel wallet** to airports, **stylish wallets** to clothing stores, and possibly have the **large wallet** spawn inside of purses. Likewise, I thought I would try to add **common wallets** to male zombies with the same frequency that purses show up on female ones. 

Still more to learn about spawn lists. Im learning most of this as I go. Other things to think about are the wallet pocket item restrictions.

I have play tested with the wallets to make sure they have the right *feel*, can fit where you would expect, and can put reasonably expected items inside. I dont think its ready yet.

File: data/json/items/generic.json [hunk 1]
```diff
@@ -3129,5 +3129,118 @@
     "symbol": ".",
     "color": "white",
     "flags": [ "NO_REPAIR" ]
+  },
```
Comment (Maleclypse) [inline data/json/items/generic.json:3132]:
```suggestion
  },
    {
    "id": "wallet",
    "type": "GENERIC",
    "category": "container",
    "name": { "str": "wallet" },
    "looks_like": "bag_plastic",
    "description": "A small, flat case used to carry items such as currency, identification documents, and laminated cards.",
    "weight": "60 g",
    "volume": "100 ml",
    "price": 1,
    "price_postapoc": 10,
    "material": [ "plastic" ],
    "pocket_data": [
      {
        "pocket_type": "CONTAINER",
        "rigid": true,
        "max_contains_weight": "600 g",
        "max_contains_volume": "500 ml",
        "moves": 400,
        "item_restriction": [
          "cash_card",
          "gasdiscount_silver",
          "gasdiscount_gold",
          "gasdiscount_platinum",
          "id_science",
          "id_military",
          "id_industrial",
          "fp_loyalty_card",
          "robofac_test_data"
        ]
      }
    ],
    "symbol": ")",
    "color": "dark_gray",
    "flags": [ "TARDIS" ]
  },
  {
    "id": "wallet_duct_tape",
    "type": "GENERIC",
    "copy-from": "wallet",
    "name": { "str": "duct tape wallet" },
    "description": "This wallet is made entirely out of duct tape.",
    "material": [ "plastic" ]
  },
  {
    "id": "wallet_leather",
    "type": "GENERIC",
    "copy-from": "wallet",
    "name": { "str": "leather wallet" },
    "description": "This wallet is made from leather.",
    "material": [ "leather" ]
  },
  {
    "id": "wallet_large",
    "type": "GENERIC",
    "copy-from": "wallet",
    "name": { "str": "large wallet" },
    "description": "A flat case used to carry items such as currency, identification documents, laminated cards, and other small items.",
    "material": [ "leather" ]
  },
  {
    "id": "wallet_stylish",
    "type": "GENERIC",
    "copy-from": "wallet",
    "name": { "str": "stylish wallet" },
    "description": "A stylish flat case used to carry items such as currency, identification documents, laminated cards, and other small items.",
    "material": [ "leather" ],
    "extend": { "flags": [ "FANCY" ] }
  },
  {
    "id": "wallet_travel",
    "type": "TOOL_ARMOR",
    "category": "clothing",
    "symbol": ";",
    "color": "dark_gray",
    "name": { "str": "travel wallet" },
    "description": "A flat case with neck strap used to carry items such as currency, identification documents, laminated cards, and other small items.",
    "price": 15000,
    "price_postapoc": 250,
    "material": [ "plastic", "leather" ],
    "flags": [ "OVERSIZE", "BELTED", "ALLOWS_NATURAL_ATTACKS", "TARDIS" ],
    "weight": "708 g",
    "volume": "500 ml",
    "bashing": 1,
    "to_hit": -1,
    "covers": [ "TORSO" ],
    "encumbrance": 2,
    "coverage": 10,
    "material_thickness": 2,
    "pocket_data": [
      {
        "pocket_type": "CONTAINER",
        "rigid": true,
        "max_contains_weight": "600 g",
        "max_contains_volume": "500 ml",
        "moves": 400,
        "item_restriction": [
          "cash_card",
          "gasdiscount_silver",
          "gasdiscount_gold",
          "gasdiscount_platinum",
          "id_science",
          "id_military",
          "id_industrial",
          "fp_loyalty_card",
          "robofac_test_data"
        ]
      }
    ]
  }
```

File: data/json/items/generic.json [hunk 2]
```diff
@@ -3139,5 +3139,135 @@
...
+    "description": "A flat case with neck strap used to carry items such as currency, identification documents, laminated cards, and other small items.",
+    "price": 15000,
+    "price_postapoc": 250,
+    "material": [ "plastic", "leather" ],
+    "flags": [ "OVERSIZE", "BELTED", "ALLOWS_NATURAL_ATTACKS", "TARDIS" ],
+    "weight": "708 g",
+    "volume": "500 ml",
+    "bashing": 1,
+    "to_hit": -1,
+    "covers": [ "TORSO" ],
```
Comment (ZhilkinSerg) [inline data/json/items/generic.json:3248]:
```suggestion
    "covers": [ "torso" ],
```

**Pull request merged**